Add library for guessing IANA timezone ids from VTIMEZONE

rsto opened the pull request:
This is a refactor from the experimental guesstz code from the cyrus-imapd/jmap-calendars-01 branch. It uses a trimmed down version of the httpd tzdist timezone truncation code to expand observances.
In contrast to the original implementation, this one uses a custom database format which is designed to allow examining timezone data directly from the mmap()ed database file.

elliefm commented:
Do the shell scripts need to be `#!/bin/bash`? Can they be `#!/bin/sh` instead?  I don't see any bash-specific syntax in there at a glance (but I didn't look closely)

Requiring bash can cause problems on some (all?) of the BSDs, where bash is historically in `/usr/bin` or maybe `/usr/local/bin`, rather than `/bin`.

If we do actually require bash we can work around this, but where it's possible to stick with POSIX sh features, it's much less nuisance to do so and make the shebang line `#!/bin/sh`. :)

rsto commented:
Thanks @elliefm, I changed the tests to use POSIX sh. Additional changes:

- added license header
- added header guards
- add the list of preferred timezone ids to the database

Passes the tests and the JMAP Calendars test suite on the jmap-calendars-01 branch.

ksmurchison reviewed (approved):
This all looks sane to me
